frontend/packages/dev-console/src/components/topology/shapes/WorkloadNode.tsx

```diff
@@ -52,7 +59,15 @@ const WorkloadNode: React.FC<NodeProps<WorkloadData>> = ({
             position={TooltipPosition.right}
           >
             <g transform={`translate(-${decoratorRadius / 2}, -${decoratorRadius / 2})`}>
-              <PenIcon style={{ fontSize: decoratorRadius }} alt="Edit Source Code" />
+              {workload.data.editUrl.includes('github.com') ? (
+                <GithubIcon style={{ fontSize: decoratorRadius }} alt="Edit Source Code" />
+              ) : workload.data.editUrl.includes('gitlab.com') ? (
+                <GitlabIcon style={{ fontSize: decoratorRadius }} alt="Edit Source Code" />
+              ) : workload.data.editUrl.includes('bitbucket.org') ? (
+                <BitbucketIcon style={{ fontSize: decoratorRadius }} alt="Edit Source Code" />
+              ) : (
+                <GitAltIcon style={{ fontSize: decoratorRadius }} alt="Edit Source Code" />
+              )}
```

Comment:
We can also use `switch` here, it will eliminate the need for so many nested conditions.

Reply:
updated as suggested, PTAL